client/src/components/layouts/components/guide/SideNav.js

```diff
@@ -27,6 +27,30 @@ class SideNav extends Component {
     this.renderParent = this.renderParent.bind(this);
   }
 
+  componentDidMount() {
```

Comment:
This logic would be better handled in a saga, much like we handle the `challengeMounted` logic.

Reply:
@Bouncey so I am trying to find the most efficient way to execute this in saga, here is what I got so far:
the saga will listen for app-mount action, and check to see if the user is in the guide by checking the window.location.pathname. if so, it will dispatch toggleExpandedState accordingly.
Since this is going to be called on every app-mount, do you think this is a good approach?